runtime/src/iree-amd-aie/driver/xrt-lite/device.cc

```diff
@@ -0,0 +1,298 @@
+                           iree_allocator_t host_allocator) {
+    IREE_ASSERT_ARGUMENT(options);
+    IREE_TRACE_ZONE_BEGIN(z0);
+
+    iree_hal_resource_initialize(&iree_hal_xrt_lite_device_vtable, &resource);
+    this->host_allocator = host_allocator;
+    shim_device = new shim_xdna::device;
+
+    iree_status_t status = iree_hal_xrt_lite_allocator_create(
+        host_allocator, shim_device, &device_allocator);
+    IREE_ASSERT(iree_status_is_ok(status));
+    iree_arena_block_pool_initialize(ARENA_BLOCK_SIZE, host_allocator,
+                                     &block_pool);
+
+    IREE_TRACE_ZONE_END(z0);
+  }
+};
+
+static iree_status_t iree_hal_xrt_lite_device_create_executable_cache(
+    iree_hal_device_t* base_device, iree_string_view_t identifier,
+    iree_loop_t loop, iree_hal_executable_cache_t** out_executable_cache) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  iree_hal_xrt_lite_device* device = IREE_HAL_XRT_LITE_CHECKED_VTABLE_CAST(
+      base_device, iree_hal_xrt_lite_device_vtable, iree_hal_xrt_lite_device);
+
+  IREE_TRACE_ZONE_END(z0);
+  return iree_hal_xrt_lite_nop_executable_cache_create(
+      device->shim_device, identifier, device->host_allocator,
+      out_executable_cache);
+}
+
+static iree_status_t iree_hal_xrt_lite_device_create_command_buffer(
+    iree_hal_device_t* base_device, iree_hal_command_buffer_mode_t mode,
+    iree_hal_command_category_t command_categories,
+    iree_hal_queue_affinity_t queue_affinity, iree_host_size_t binding_capacity,
+    iree_hal_command_buffer_t** out_command_buffer) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  if (!iree_all_bits_set(mode, IREE_HAL_COMMAND_BUFFER_MODE_ONE_SHOT)) {
+    IREE_TRACE_ZONE_END(z0);
+    return iree_make_status(IREE_STATUS_UNIMPLEMENTED,
+                            "unimplmented multi-shot command buffer");
+  }
+
+  iree_hal_xrt_lite_device* device = IREE_HAL_XRT_LITE_CHECKED_VTABLE_CAST(
+      base_device, iree_hal_xrt_lite_device_vtable, iree_hal_xrt_lite_device);
+
+  IREE_TRACE_ZONE_END(z0);
+  return iree_hal_deferred_command_buffer_create(
+      device->device_allocator, mode, command_categories, binding_capacity,
+      &device->block_pool, device->host_allocator, out_command_buffer);
+}
+
+static iree_status_t iree_hal_xrt_lite_device_create_semaphore(
+    iree_hal_device_t* base_device, uint64_t initial_value,
+    iree_hal_semaphore_flags_t flags, iree_hal_semaphore_t** out_semaphore) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  iree_hal_xrt_lite_device* device = IREE_HAL_XRT_LITE_CHECKED_VTABLE_CAST(
+      base_device, iree_hal_xrt_lite_device_vtable, iree_hal_xrt_lite_device);
+
+  IREE_TRACE_ZONE_END(z0);
+  return iree_hal_xrt_lite_semaphore_create(device->host_allocator,
+                                            initial_value, out_semaphore);
+}
+
+static iree_status_t iree_hal_xrt_lite_device_queue_execute(
+    iree_hal_device_t* base_device, iree_hal_queue_affinity_t queue_affinity,
+    const iree_hal_semaphore_list_t wait_semaphore_list,
```

Comment:
Are we supposed to use the "wait_semphore_list" or "signal_semaphore_list" for telling the host when the execution result is available?

Reply:
Please consult consult [iree/runtime/src/iree/hal/semaphore.c](https://github.com/iree-org/iree/blob/a28f76fdf3661196e30f4720c29ee4e2427ad415/runtime/src/iree/hal/semaphore.c).